lib/scholar/metrics/classification.ex

```diff
@@ -586,86 +589,174 @@ defmodule Scholar.Metrics.Classification do
   end
 
   @doc """
-  Calculates F1 score given rank-1 tensors which represent
+  Calculates F-beta score given rank-1 tensors which represent
   the expected (`y_true`) and predicted (`y_pred`) classes.
 
   If all examples are true negatives, then the result is 0 to
   avoid zero division.
 
   ## Options
 
-  #{NimbleOptions.docs(@f1_score_schema)}
+  #{NimbleOptions.docs(@fbeta_score_schema)}
 
   ## Examples
 
       iex> y_true = Nx.tensor([0, 1, 1, 1, 1, 0, 2, 1, 0, 1], type: :u32)
       iex> y_pred = Nx.tensor([0, 2, 1, 1, 2, 2, 2, 0, 0, 1], type: :u32)
-      iex> Scholar.Metrics.Classification.f1_score(y_true, y_pred, num_classes: 3)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(1), num_classes: 3)
       #Nx.Tensor<
         f32[3]
         [0.6666666865348816, 0.6666666865348816, 0.4000000059604645]
       >
-      iex> Scholar.Metrics.Classification.f1_score(y_true, y_pred, num_classes: 3, average: :macro)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(2), num_classes: 3)
+      #Nx.Tensor<
+        f32[3]
+        [0.6666666865348816, 0.5555555820465088, 0.625]
+      >
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(0.5), num_classes: 3)
+      #Nx.Tensor<
+        f32[3]
+        [0.6666666865348816, 0.8333333134651184, 0.29411765933036804]
+      >
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(2), num_classes: 3, average: :macro)
       #Nx.Tensor<
         f32
-        0.5777778029441833
+        0.6157407760620117
       >
-      iex> Scholar.Metrics.Classification.f1_score(y_true, y_pred, num_classes: 3, average: :weighted)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(2), num_classes: 3, average: :weighted)
       #Nx.Tensor<
         f32
-        0.6399999856948853
+        0.5958333611488342
       >
-      iex> Scholar.Metrics.Classification.f1_score(y_true, y_pred, num_classes: 3, average: :micro)
+      iex> Scholar.Metrics.Classification.fbeta_score(y_true, y_pred, Nx.tensor(0.5), num_classes: 3, average: :micro)
       #Nx.Tensor<
         f32
         0.6000000238418579
       >
-      iex> Scholar.Metrics.Classification.f1_score(Nx.tensor([1,0,1,0]), Nx.tensor([0, 1, 0, 1]), num_classes: 2, average: :none)
+      iex> Scholar.Metrics.Classification.fbeta_score(Nx.tensor([1, 0, 1, 0]), Nx.tensor([0, 1, 0, 1]), Nx.tensor(0.5), num_classes: 2, average: :none)
       #Nx.Tensor<
         f32[2]
         [0.0, 0.0]
       >
   """
-  deftransform f1_score(y_true, y_pred, opts \\ []) do
-    f1_score_n(y_true, y_pred, NimbleOptions.validate!(opts, @f1_score_schema))
+  deftransform fbeta_score(y_true, y_pred, beta, opts \\ []) do
```

Comment:
You shouldn't rename the function because this is a breaking change. Either keep the name, or add a separate function and delegate both to the same private implementation.

I'd rather keep the name because they're both basically the same function.

Reply:
It is not being renamed, please check below the diff :D